lib/Sema/ConstraintSystem.h

```diff
@@ -1092,7 +1061,7 @@ class ConstraintSystem {
   llvm::FoldingSetVector<ConstraintLocator> ConstraintLocators;
 
   /// The overload sets that have been resolved along the current path.
-  ResolvedOverloadSetListItem *resolvedOverloadSets = nullptr;
+  llvm::MapVector<ConstraintLocator *, SelectedOverload> ResolvedOverloads;
```

Comment:
Since we are using `SelectedOverload` everywhere now maybe it makes sense to attach locator directly to it and use `SetVector` here which would also allow us to return `ArrayRef` from `getResolvedOverloads` method?

Reply:
@xedin Hmm, I might be missing something, but wouldn't a `SetVector` prevent us from looking up an overload for a given locator?

Reply:
Ah, sorry you are right, for some reason I thought that we only needed uniqueness/existence checks :) The main reason why I suggest that is because I don't like the fact that `getResolvedOverloads` returns a copy now.

Reply:
@xedin Ah I see! I guess really `MapVector` should also have a `getArrayRef()` member. In the mean time though, how about we return a const reference to the map vector from `getResolvedOverloads`?

Reply:
Sure, sounds good!